crates/anvil/src/eth/api.rs

```diff
@@ -2230,84 +2230,43 @@ impl EthApi {
                 let allowance = available_funds.checked_div(gas_price).unwrap_or_default();
                 if highest_gas_limit > allowance {
                     trace!(target: "node", "eth_estimateGas capped by limited user funds");
-                    highest_gas_limit = allowance;
+                    highest_gas_limit = std::cmp::min(highest_gas_limit, allowance);
```

Comment:
this min is redundant because allowance is smaller here,
we could replace the entire if with this and remove the trace, not that useful